client/src/components/Map/components/Block.js

```diff
@@ -56,13 +59,15 @@ export class Block extends Component {
     this.renderChallenges = this.renderChallenges.bind(this);
   }
 
-  handleBlockClick() {
-    const { blockDashedName, toggleBlock } = this.props;
-    ga.event({
-      category: 'Map Block Click',
-      action: blockDashedName
-    });
-    return toggleBlock(blockDashedName);
+  handleBlockClick(isOpen) {
+    return () => {
+      const { blockDashedName, toggleBlock } = this.props;
+      ga.event({
+        category: 'Map Block Click',
+        action: blockDashedName
+      });
+      return toggleBlock(blockDashedName, !isOpen);
```

Comment:
`toggleBlock` didn't need to know the current expanded state before. I would be wary of this and ask why it needed to be changed.

Stateless is easier to maintain.